src/controllers/GatewayController.ts

```diff
@@ -150,4 +156,29 @@ export default class GatewayController {
 		if (!userConfig) throw new GatewayError('Not authenticated');
 		this.discoveryQueue.removeFromQueue(userConfig.id);
 	}
+
+	public async onTyping(ws: WebSocket, packet: ClientTypingPacket) {
+		const userConfig = this.authenticatedClients.get(ws);
+		if (!userConfig) throw new GatewayError('Not authenticated');
+
+		const { channelID } = packet.data;
+		const channel = await this.channelService.findOne({ id: channelID });
```

Comment:
Ah also, if a user provides no channelID, this would actually resolve with any channel which is a security issue. Can you do a null check on channelID before this line just to prevent that from happening?